memcache/memcache.go

```diff
@@ -258,8 +260,10 @@ func (c *Client) dial(addr net.Addr) (net.Conn, error) {
 		cn  net.Conn
 		err error
 	}
-
-	nc, err := net.DialTimeout(addr.Network(), addr.String(), c.netTimeout())
+	if c.DialTimeout == nil {
```

Comment:
@themihai what do you think about @bboreham proposal below?
This proposal is already implemented in grafana fork of gomemcache: https://github.com/grafana/gomemcache/pull/2/files

```suggestion
        dialTimeout := c.DialTimeout
	if dialTimeout == nil {
		dialTimeout = net.DialTimeout
	}
	nc, err := dialTimeout(addr.Network(), addr.String(), c.netTimeout())
```
What do you think about adopting it here?
@bboreham I hope you don't mind I just copied your code for suggestion :)